### PR TITLE
Improve bundle / chunk sizes

### PR DESCRIPTION
**Pull request by jschuler**
### Tldr;
| Type | Current PF5 branch | This branch |
|---|---|---|
| JS | PF4+PF5 **2.8MB** |  PF4 **848KB** + PF5 **1.26MB** |
| CSS | PF4+PF5+Local **3.8MB** + **2.7MB** | PF4 **1.9MB** + PF5 **1.8MB** + Local **775KB** |

===========================

Some improvements to the webpack config to reduce size of chunks and CSS bundles.
Combined efforts with @vojtechszocs, pulled his changes into this PR as well.

**This is the current PF5 bundle that we are improving with this PR:**
Screenshot 2023-11-09 at 4 08 19 PM
- All PF4 and PF5 packages are in a single JS chunk **2.8MB**
- CSS bundles are **3.8MB** and **2.7MB**, they contain local, PF4, PF5, extension styles with duplication across

**After this PR's changes, and also comparing it to PF4 master branch:**

| PF4 (master) | PF5 (branch) |
|---|---|
| Screenshot 2023-11-09 at 1 02 46 AM | Screenshot 2023-11-09 at 12 52 12 AM |

Screenshot 2023-11-09 at 12 49 51 AM

- Separate smaller JS chunks for PF4 **848KB** and PF5 **1.26MB**
- PF4 shared CSS (also contains utility styles) for dynamic plugins: **1.9MB**
- PF5 CSS: **1.8MB**
- Local (main) styles: **775KB**

**Summary**
After these changes, compared to PF4 master we roughly these extra things to load (and cache):
- PF4 CSS bundle increased slightly since we bundle PF4 CSS + utilities for dynamic plugins **1.5MB -> 1.9MB**
- PF5 CSS **1.8MB**
- PF5 JS chunk **1.26MB**

Compared to the PF5 PR branch we are merging into:
- PF4+PF5 JS **2.8MB** (1 chunk) -> **2.1MB** (2 chunks)
- PF4+PF5+Local CSS **6.5MB** (2 bundles) -> **4.48MB** (3 bundles)

### CODE REVIEW

File: frontend/package.json [hunk 1]
```diff
@@ -137,8 +137,10 @@
     "@patternfly/quickstarts": "5.1.0",
     "@patternfly/react-catalog-view-extension": "5.0.0",
     "@patternfly/react-charts": "7.1.0",
+    "@patternfly/react-console": "5.0.0",
```
Comment (jschuler) [inline frontend/package.json:140]: @patternfly/react-console and @patternfly/react-styles were missed and still at version 4.x under `node_modules/@patternfly/`. This improved CSS bundle sizes.

File: frontend/public/vendor.scss [hunk 2]
```diff
@@ -27,10 +27,5 @@
 @import '~@patternfly/patternfly/patternfly-charts';
 @import '~@patternfly/patternfly/patternfly-addons';
 
-// PatternFly 4 stylesheets. We need these to support existing Console dynamic plugins
-// that may use PatternFly 4 shared modules as their runtime dependencies.
-@import '~@patternfly-4/patternfly/patternfly';
```
Comment (jschuler) [inline frontend/public/vendor.scss:32]: pulled these out into a separate file for better CSS bundle separation

File: frontend/webpack.config.ts [hunk 3]
```diff
@@ -34,7 +35,9 @@ const WDS_PORT = 8080;
 /* Helpers */
 const extractCSS = new MiniCssExtractPlugin({
   filename:
-    NODE_ENV === 'production' ? 'app-bundle.[contenthash].css' : 'app-bundle.[name].[hash].css',
+    NODE_ENV === 'production'
+      ? 'app-bundle.[name].[contenthash].css'
```
Comment (jschuler) [inline frontend/webpack.config.ts:39]: Need the `[name]` to identify later when we want to exclude certain chunks from the HTML

File: frontend/webpack.config.ts [hunk 4]
```diff
@@ -46,20 +49,30 @@ const getVendorModuleRegExp = (vendorModules: string[]) =>
...
-const sharedPluginTest = getVendorModuleRegExp(
-  sharedPluginModules.map((moduleName) =>
-    // Console provides specific PatternFly 4 shared modules to its dynamic plugins.
-    // These are represented in Console webpack compilation as @patternfly-4/* modules.
-    moduleName.startsWith('@patternfly/')
-      ? moduleName.replace(/^@patternfly\//, '@patternfly-4/')
-      : moduleName,
-  ),
-);
+const sharedPluginModulesTest = getVendorModuleRegExp(
```
Comment (jschuler) [inline frontend/webpack.config.ts:52]: @vojtechszocs updated this function

File: frontend/webpack.config.ts [hunk 5]
```diff
@@ -46,20 +49,30 @@ const getVendorModuleRegExp = (vendorModules: string[]) =>
...
+
+    return moduleName;
+  }),
+);
 
 const config: Configuration = {
-  entry: ['./public/components/app.jsx', 'monaco-editor-core/esm/vs/editor/editor.worker.js'],
+  entry: {
+    main: ['./public/components/app.jsx', 'monaco-editor-core/esm/vs/editor/editor.worker.js'],
+    'vendor-patternfly-4-shared': './public/vendor-patternfly-4-shared.scss',
```
Comment (jschuler) [inline frontend/webpack.config.ts:74]: Separate entry point to include PF4 CSS

**Review by jpuzz0 — APPROVED** [review]
